docs/builds.rst

```diff
@@ -0,0 +1,328 @@
+.. _Builds:
+
+Builds
+======
+
+To ensure reproducibility of experiments, it is possible to let simexpal pull programs from a VCS (currently
+only Git is supported) and build them automatically. To achieve this, we need to specify a ``builds`` and
+``revisions`` stanza in the ``experiments.yml``.
+
+On this page we will explain the ``builds`` key and mainly use the
+`C++ example <https://github.com/hu-macsy/simexpal/tree/master/examples/sorting_cpp>`_ from the
+:ref:`QuickStart` guide in order to do so. For example, we will explain how to state a Git repository, enable
+the automated build support and the directories involved in the build process. Furthermore we will present
+additional options such as pulling Git submodules and setting additional environment variables for the build process.
+
+You can find information regarding the ``revisions`` key on the :ref:`Revisions` page.
+
+Stating a Git Repository
+------------------------
+
+To state our git repository, we use the keys
+
+- ``name``: (arbitrary) name of the build
+- ``git``: link to the git-repository
+
+as follows:
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to state a git-repository in the experiments.yml file.
+
+    builds:
+      - name: simexpal
+        git: https://github.com/hu-macsy/simexpal
+
+.. _PullingGitSubmodules:
+
+Pulling Git Submodules
+----------------------
+
+If your program contains Git submodules you need to set the
+
+- ``recursive-clone``: boolean (``True``/``False``) - whether to pull git submodules recursively or not
+
+key to ``True`` in order for simexpal to pull the respective files.
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to let simexpal pull Git submodules in the experiments.yml file.
+
+    builds:
+      - name: networkit
+        git: https://github.com/networkit/networkit
+        recursive-clone: True
+
+If ``recursive-clone`` is not set, it will default to ``False``.
+
+Automated Builds
+----------------
+
+To enable automated build support, we can specify the following keys:
+
+- ``regenerate``: list of dictionaries containing regeneration parameters
+- ``configure``: list of dictionaries containing configuration parameters
+- ``compile``: list of dictionaries containing compilation parameters
+- ``install``: list of dictionaries containing installation parameters
```

Comment:
Maybe explain when these steps run / what they typically do (regenerate: prepare the source directory before the build starts, e.g., by downloading additional dependencies or subprojects. configure: run `cmake` or `./configure` script.).

Reply:
I added some small paragraphs regarding the points you mentioned.